packages/alfa-rules/src/sia-r66/rule.ts

```diff
@@ -35,31 +37,67 @@ export default Rule.Atomic.of<Page, Text, Question.Metadata>({
   evaluate({ device, document }) {
     return {
       applicability() {
-        return visit(document);
+        let disabledWidgetTexts: Set<Text> = Set.empty();
+        let textNodes: Sequence<Text> = Sequence.empty();
+        gather(document);
+        return getApplicableTexts();
 
-        function* visit(node: Node): Iterable<Text> {
+        function gather(node: Node): void {
+          // Gather all aria-disabled widgets on the document
           if (
             test(
               and(
                 isElement,
                 or(
-                  not(hasNamespace(Namespace.HTML)),
                   hasRole(device, (role) => role.isWidget()),
-                  and(hasRole(device, "group"), isSemanticallyDisabled)
+                  isSemanticallyDisabled
                 )
```

Comment:
This should be a `and` (to get disabled widgets: both widget and disabled).
So
```html
<label>
  some text
  <input />
</label>
```
the text is in the name of an active widget and should be tested.